core/environment/environment.go

```diff
@@ -439,7 +478,49 @@ func newEnvironment(userVars map[string]string, newId uid.ID) (env *Environment,
 							Infof("auto stop transition scheduled, expected execution at %s", expected)
 					}
 
+					runStartCompletionTime := time.Now()
+					runStartCompletionTimeS := strconv.FormatInt(runStartCompletionTime.UnixMilli(), 10)
+					env.workflow.SetRuntimeVar("run_start_completion_time_ms", runStartCompletionTimeS)
+					env.workflow.SetRuntimeVar("run_end_completion_time_ms", "") // we delete previous EOEOR
+
+					runEvent := &pb.Ev_RunEvent{
+						EnvironmentId:    envId.String(),
+						RunNumber:        env.GetCurrentRunNumber(),
+						State:            env.Sm.Current(),
+						Error:            "",
+						Transition:       e.Event,
+						TransitionStatus: pb.OpStatus_DONE_OK,
+						Vars:             nil,
+					}
+					if e.Err != nil {
+						runEvent.Error = e.Err.Error()
+						runEvent.TransitionStatus = pb.OpStatus_DONE_ERROR
+					}
+
+					the.EventWriterWithTopic(topic.Run).WriteEventWithTimestamp(runEvent, runStartCompletionTime)
+
 				} else if e.Event == "STOP_ACTIVITY" {
+					runEndCompletionTime := time.Now()
+					runEndCompletionTimeS := strconv.FormatInt(runEndCompletionTime.UnixMilli(), 10)
+					env.workflow.SetRuntimeVar("run_end_completion_time_ms", runEndCompletionTimeS)
+					env.workflow.SetRuntimeVar("run_start_completion_time_ms", "") // we delete previous EOSOR
```

Comment:
Why delete it already? I would not exclude that we could have a plugin which wants to retrieve the 4 timestamps only at `after_STOP_ACTIVITY`.

At least we could be coherent and delete either all the previous timestamps (SOSOR, EOSOR, SOEOR, EOEOR) or (perhaps better) keep them until the next `before_START_ACTIVITY`.